src/Caller.cs

```diff
@@ -21,6 +23,61 @@ internal Caller(IntPtr handle)
             this.store = context.Store;
         }
 
+        /// <summary>
+        /// Gets a Span from an exported memory of the caller by the given name.
+        /// </summary>
+        /// <param name="name">The name of the exported memory.</param>
+        /// <param name="address">The zero-based address of the start of the span.</param>
+        /// <param name="length">The length of the span.</param>
+        /// <param name="result">The Span of memory (if the function returns true)</param>
+        /// <returns>Returns true if the exported memory is found or false if a memory of the requested name is not exported.</returns>
+        /// <remarks>
+        /// <para>
+        /// The span may become invalid if the memory grows.
+        ///
+        /// This may happen if the memory is explicitly requested to grow or
+        /// grows as a result of WebAssembly execution.
+        /// </para>
+        /// <para>
+        /// Therefore, the returned span should not be used after calling the grow method or
+        /// after calling into WebAssembly code.
+        /// </para>
+        /// <para>
+        /// Note that WebAssembly always uses little endian as byte order. On platforms 
+        /// that use big endian, you will need to convert numeric values accordingly.
+        /// </para>
+        /// </remarks>
+        public bool TryGetMemorySpan<T>(string name, long address, int length, out Span<T> result)
+            where T : unmanaged
+        {
+            var nameLength = Encoding.UTF8.GetMaxByteCount(name.Length);
```

Comment:
It seems you reverted back to the previous approach of using `GetMaxByteCount` when doing the force-push, was this deliberate? (Sorry for not catching this earlier).
Thanks!

Reply:
No it wasn't! I'm not sure quite how I managed that, I'll put in a PR to fix it now. Thanks for catching it 👍